onnx/test/new_version_converter_test.py

```diff
@@ -0,0 +1,1005 @@
+        output_shapes=[[3, 4, 5]],  # type: List[List[Optional[int]]]
+        input_types=None,  # type: Union[List[Any], None]
+        output_types=None,  # type: Union[List[Any], None]
+        initializer=[],  # type: List[Any]
+        attrs={},  # type: Dict[Text, Any]
+        seq_inputs=[],  # type: List[int]
+        seq_outputs=[],  # type: List[int]
+        to_opset=None  # type: Union[int, None]
+    ):  # type: (...) -> None
+        global tested_ops
+        tested_ops.append(op)
+
+        n_inputs = len(input_shapes)
+        letters = list(string.ascii_lowercase)[:n_inputs]
+        input_names = [
+            letter if shape != '' else '' for (letter, shape) in zip(letters, input_shapes)
+        ]
+        if input_types is None:
+            input_types = [TensorProto.FLOAT] * n_inputs
+        is_sequence = [0 if id not in seq_inputs else 1 for id in range(n_inputs)]
+        # turn empty strings into [0] to ease type analysis, even though those entries
+        # will be ignored
+        input_shapes_cast = cast(List[List[int]],
+                [[0] if isinstance(shape, str) else shape for shape in input_shapes]
+        )
+        inputs = [
+            helper.make_tensor_value_info(name, ttype, shape) if is_sequence == 0
+            else helper.make_tensor_sequence_value_info(name, ttype, shape)
+            for (name, ttype, shape, is_sequence)
+            in zip(input_names, input_types, input_shapes_cast, is_sequence) if name != ''
+        ]
+
+        n_outputs = len(output_shapes)
+        output_names = list(string.ascii_lowercase)[n_inputs:n_inputs + n_outputs]
+        if output_types is None:
+            output_types = [TensorProto.FLOAT] * n_outputs
+        is_sequence = [0 if id not in seq_outputs else 1 for id in range(n_outputs)]
+        output_shapes_cast = cast(List[List[int]],
+                [[0] if isinstance(shape, str) else shape for shape in output_shapes]
+        )
+        outputs = [
+            helper.make_tensor_value_info(name, ttype, shape) if is_sequence == 0
+            else helper.make_tensor_sequence_value_info(name, ttype, shape)
+            for (name, ttype, shape, is_sequence)
+            in zip(output_names, output_types, output_shapes_cast, is_sequence)
+        ]
+
+        node = helper.make_node(op, input_names, output_names, **attrs)
+        graph = helper.make_graph([node], op, inputs, outputs, initializer)
+        original = helper.make_model(
+            graph,
+            producer_name='test',
+            opset_imports=[helper.make_opsetid('', from_opset)]
+        )
+        onnx.checker.check_model(original)
+        shape_inference.infer_shapes(original)
+
+        if to_opset is None:
+            to_opset = latest_opset
+
+        converted = version_converter.convert_version(original, to_opset)
+        onnx.checker.check_model(converted)
+        shape_inference.infer_shapes(converted)
```

Comment:
To strengthen the check, shall we use a stricter shape_inference? like `shape_inference.infer_shapes(converted, strict_mode=True)`

Reply:
Good point, added it